go/libkb/session.go

```diff
@@ -285,13 +287,29 @@ func (s *Session) check() error {
 		}
 	} else {
 		s.G().Log.Notice("Stored session expired")
-		s.valid = false
+		s.Invalidate()
 	}
 
 	s.G().Log.Debug("- Checked session")
 	return nil
 }
 
+// Invalidate marks the session as invalid and posts a logout
+// notification.
+func (s *Session) Invalidate() {
+	s.G().Log.Debug("invalidating session")
+	s.clearState()
+	s.G().NotifyRouter.HandleLogout()
```

Comment:
Does this get called from global logout? If so we will send two handle logout notifications. We should get rid of the notification in globals.